src/compiler/scala/tools/nsc/backend/jvm/opt/InlinerHeuristics.scala

```diff
@@ -64,7 +64,7 @@ class InlinerHeuristics[BT <: BTypes](val bTypes: BT) {
     }).filterNot(_._2.isEmpty).toMap
   }
 
-  private def isTraitStaticSuperAccessorName(s: String) = s.endsWith("$")
+  private def isTraitStaticSuperAccessorName(s: String) = s != "$init$" && s.endsWith("$")
```

Comment:
Do we emit any other static methods (e.g. static-ified lambda lifted or lambda impl methods) that could possible have the trailing `$` and also be misclassified?